rodan/jobs/helloworld/helloworld.py

```diff
@@ -29,6 +29,45 @@ def run_my_task(self, inputs, settings, outputs):
         outfile.close()
         return True
 
+class HelloWorldMultiPort(RodanTask):
+    name = 'Hello World Multiple Ports'
+    author = 'Studio theYANG'
+    description = 'Concatenate all input files and append "Hello World MultiPort"'
+    settings = {}
+    enabled = True
+    category = "Test"
+    interactive = False
+
+    input_port_types = (
+        {
+            'name': 'Text input',
+            'minimum': 0,
+            'maximum': 10,
+            'resource_types': ['text/plain']
+        },
+    )
+    output_port_types = (
+        {
+            'name': 'Text output',
+            'minimum': 1,
+            'maximum': 10,
+            'resource_types': ['text/plain']
+        },
+    )
+
+    def run_my_task(self, inputs, settings, outputs):
+        concatenated = ""
+        for input_type in inputs:
+            for input in inputs[input_type]:
+                with open(input["resource_path"], "r") as infile:
+                    concatenated += infile.read() + "\n"
+
+        for output_type in outputs:
+            for output in outputs[output_type]:
+                with open(output["resource_path"], "w") as outfile:
+                    outfile.write(concatenated)
+                    outfile.write("Hello World MultiPort")
+
```

Comment:
Tests can also be placed in the job by defining the "test_my_task" method. 

```python
def test_my_task(self, testcase):
  inputs = { 'in': [{'resource_type': 'text/plain', 'resource_path': testcase.new_available_path()}]}
  ...
  self.run_my_task(inputs, {}, outputs)
  # open output
  result = outputs['out'][0]['resource_path']) 

  testcase.assertEqual(result, "whatever you expect to test") 
  ...
``` 
I wonder if writing specific tests in their own jobs would not be better instead of writing them with model mommy. What are your thoughts, @lingxiaoyang

Reply:
I've discussed it with others. Going forward we will write tests in the jobs themselves instead of being a part of Rodan tests. This will be fine for now, but will change in the near future. Thank you.

Reply:
Sorry, I just saw this!

Yes, agreeing with you, I'm missing the test method here. My omission.